main.go

```
@@ -85,35 +87,38 @@ func printResp(resp *graph.ExecutionResponse, duration time.Duration) {
 // Loop the request util fatal or timeout
 // We treat one line as one query
 // Add line break yourself as `SHOW \<CR>HOSTS`
-func loop(client *ngdb.GraphClient, c cli.Cli) error {
+func loop(client *ngdb.GraphClient, c cli.Cli) {
 	for {
 		line, err, exit := c.ReadLine()
-		lineString := string(line)
 		if exit {
-			return err
+			return
 		}
-		if len(line) == 0 {
+		if err == nil {
+			if len(line) == 0 {
+				continue
+			}
+			// Client Side command
+			if clientCmd(line) {
+				// Quit
+				break
+			}
+			start := time.Now()
+			resp, err := client.Execute(line)
+			duration := time.Since(start)
+			if err != nil {
+				log.Fatalf("Execute error, %s", err.Error())
+			}
+			printResp(resp, duration)
+			fmt.Println(time.Now().In(time.Local).Format(time.RFC1123))
 			fmt.Println()
-			continue
-		}
-
-		// Client side command
-		if clientCmd(lineString) {
-			// Quit
-			return nil
-		}
-
-		start := time.Now()
-		resp, err := client.Execute(lineString)
-		duration := time.Since(start)
-		if err != nil {
-			log.Fatalf("Execute error, %s", err.Error())
+			c.SetSpace("(none)")
+			if len(string(resp.SpaceName)) > 0 {
+				c.SetSpace(string(resp.SpaceName))
+			}
+		} else {
+			log.Print("err:", err)
+			break
```

Don't ignore this error

---

I don't understand, how to deal with the error?